src/NUnitFramework/framework/Internal/Tests/Test.cs

```diff
@@ -349,6 +349,16 @@ public void ApplyAttributesToTest(ICustomAttributeProvider provider)
         }
 #endif
 
+        /// <summary>
+        /// Mark the test as Invalid (not runnable) specifying a reason
+        /// </summary>
+        /// <param name="reason">The reason the test is not runnable</param>
+        public void MakeInvalid(string reason)
+        {
+            RunState = RunState.NotRunnable;
+            Properties.Add(PropertyNames.SkipReason, reason);
```

Comment:
I like this. Just one thing, maybe we should skip the`Properties.Add` if reason is null or empty, or else throw if it should be required?